### PR TITLE
Enhanced error handling

### PR DESCRIPTION
**Pull request by alejandro-gonzalez-scale**
## Description

This pull request introduces several improvements and enhancements to the existing Python script. The key changes are outlined below:

### API Key Validation

- In the `Api` class constructor (`__init__` method), we now validate the provided API key to ensure it is not empty or `None`. If the key is invalid, a `ScaleException` is raised with the message "Please provide a valid API Key."

### Enhanced Error Handling

- The error handling logic has been improved in the `_api_request` method. Specifically, when encountering a 409 HTTP status code and specific conditions are met (endpoint contains "task" and `body` contains a "unique_id"), the code checks for retry history and performs additional API requests to retrieve task data if necessary.

### Readability and Maintenance

- The code has been refactored to enhance readability and maintainability. Descriptive variable names have been used, and comments have been added to explain each step of the process.

### Version Update

- The version number in the `_version.py` file has been updated from "2.14.4" to "2.14.5," and the package name remains "scaleapi."

### Miscellaneous

- Minor adjustments have been made to variable naming and code structure to align with Python naming conventions and improve overall code quality.

## How to Test

To test these changes, you can use the modified script in your existing project or application. Be sure to provide a valid API key when initializing the `Api` class.

## Additional Information

- The script remains backward-compatible with the original version.
- These changes are designed to enhance error handling and make the code more robust when dealing with specific API responses.
- We have maintained consistency with the existing coding style and conventions.

Please review and merge this pull request to incorporate these improvements into the project.

### CODE REVIEW

File: scaleapi/api.py [hunk 1]
```diff
@@ -145,34 +123,34 @@ def _api_request(
...
+        
+                # Perform a GET request to retrieve task data
+                new_res = self._http_request("GET", new_url, headers=headers, auth=auth)
+
+                if new_res.status_code == 200:
+                    new_res_data = new_res.json()
+                    if new_res_data["docs"]:
+                        json = new_res_data["docs"][0]
+                    else:
+                        self._raise_on_response(new_res)
```
Comment (jonathanfeng-scale) [inline scaleapi/api.py:139]:
```
self._raise_on_response(new_res)
```

this function doesn't exist, the correct function is actually mis-spelled lol

**Review by jonathanfeng-scale — CHANGES_REQUESTED** [review]

File: scaleapi/api.py (hunk 1 above)
Comment (alejandro-gonzalez-scale) [reply scaleapi/api.py]: spell check, lol
reverted back to `self._raise_on_respose(res)`

**Review by jonathanfeng-scale — APPROVED** [review]

**Review by jonathanfeng-scale — APPROVED** [review]

File: scaleapi/api.py [hunk 2]
```diff
@@ -1,27 +1,20 @@
...
 HTTP_RETRY_ALLOWED_METHODS = frozenset({"GET", "POST", "DELETE"})
-
-
 class Api:
     """Internal Api reference for handling http operations"""
-
     def __init__(
         self,
-        api_key,
+        api_key,
```
Comment (fatihkurtoglu) [inline scaleapi/api.py:17]: nit: I feel like removing those new lines makes code a bit hard to read, wdyt?

File: scaleapi/_version.py [hunk 3]
```diff
@@ -1,2 +1,2 @@
-__version__ = "2.14.4"
+__version__ = "2.14.5"
```
Comment (fatihkurtoglu) [inline scaleapi/_version.py:1]: I think you can even make it 2.15.0  this looks like a relatively major improvement to the SDK!

File: scaleapi/api.py [hunk 4]
```diff
@@ -30,7 +30,7 @@ def __init__(
     ):
         if api_key == "" or api_key is None:
             raise ScaleException("Please provide a valid API Key.")
-
+
```
Comment (fatihkurtoglu) [inline scaleapi/api.py]: Those extra tabs/spaces are generally auto-removed when you have the `black` package in your local (or via pre-commit package). Do you use them locally?

**Comment by fatihkurtoglu** [comment]
Also interesting that CircleCI tests are not running 🤔 we also need to check that.

**Comment by fatihkurtoglu** [comment]
oh I know why, I think this PR is on a fork in your repo, maybe that's why.

**Comment by fatihkurtoglu** [comment]
I think you need to request write access to this repo from Opal, then can we make the same PR from a branch by any chance? I think CI doesn't run on other forked repos (that's my assumption)

File: scaleapi/api.py (hunk 2 above)
Comment (alejandro-gonzalez-scale) [reply scaleapi/api.py]: Yea, fixing the local formatting issue.

File: scaleapi/_version.py (hunk 3 above)
Comment (alejandro-gonzalez-scale) [reply scaleapi/_version.py]: Sounds good, I've updated the version to 2.15.0

**Pull request closed without merging**